crates/trigger-redis/src/lib.rs

```diff
@@ -99,7 +99,11 @@ impl TriggerExecutor for RedisTrigger {
             .collect();
 
         // wait for the first handle to be returned and drop the rest
-        let (_, _, rest) = futures::future::select_all(tasks).await;
+        let (result, _, rest) = futures::future::select_all(tasks).await;
+        if let Err(err) = result? {
+            eprintln!("Error: {err:?}");
```

Comment:
Prefer `terminal::error!` - it will provide you with Spin-standard colourisation.

Alternatively, and perhaps simpler, return the error from `run`.  That should bubble up to the top level, so Rust will output it and produce the appropriate exit code.

Reply:
I like this better! will update.